src/App.tsx

```diff
@@ -17,18 +45,27 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter filteredTodos={setFilteredTodos} todos={todos} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading && todos.length > 0 && (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodo={selectedTodo}
+                  selectTodo={(todo: Todo | null) =>
+                    todo ? selectTodo(todo) : null
+                  }
```

Comment:
Create separate function for this, or move this logic inside `selectTodo` function